superset-frontend/src/components/Menu/Menu.test.tsx

```diff
@@ -157,43 +159,63 @@ test('should render all the top navbar menu items', () => {
   } = mockedProps;
   render(<Menu {...mockedProps} />);
   menu.forEach(item => {
-    const menuItem = screen.getByText(item.label);
-    expect(menuItem).toHaveAttribute('href', item.url);
+    expect(screen.getByText(item.label)).toBeInTheDocument();
   });
 });
 
-test('should render the top navbar child menu items', () => {
+test('should render the top navbar child menu items', async () => {
   const {
     data: { menu },
   } = mockedProps;
   render(<Menu {...mockedProps} />);
-  const datasets = screen.getByText('Datasets');
-  const databases = screen.getByText('Databases');
-  const dataset = menu[1].childs![0] as { url: string };
-  const database = menu[1].childs![2] as { url: string };
+  const sources = screen.getByText('Sources');
+  userEvent.hover(sources);
+  setTimeout(() => {
+    const datasets = screen.getByText('Datasets');
+    const databases = screen.getByText('Databases');
+    const dataset = menu[1].childs![0] as { url: string };
+    const database = menu[1].childs![2] as { url: string };
 
-  expect(datasets).toHaveAttribute('href', dataset.url);
-  expect(databases).toHaveAttribute('href', database.url);
+    expect(datasets).toHaveAttribute('href', dataset.url);
+    expect(databases).toHaveAttribute('href', database.url);
+  }, 1000);
+});
+
+test('should render the dropdown items', async () => {
+  render(<Menu {...mockedProps} />);
+  dropdownItems.forEach(item => {
+    setTimeout(() => {
+      expect(screen.getByText(item.label)).toHaveAttribute('href', item.url);
+      expect(screen.getByTestId(`menu-item-${item.label}`)).toBeInTheDocument();
+    }, 10);
+  });
 });
 
 test('should render the Settings', () => {
```

Comment:
Instead of the `setTimeout` we can use `findBy` like:

```
test('should render the Settings', async () => {
  render(<Menu {...mockedProps} />);
  const settings = await screen.findByText('Settings');
  expect(settings).toBeInTheDocument();
});
```